app/views/govuk_publishing_components/components/_title.html.erb

```diff
@@ -24,7 +21,7 @@
 <%= content_tag(:div, class: classes) do %>
   <% if context %>
     <span class="govuk-caption-xl gem-c-title__context" <%= "lang=#{context_locale}" if context_locale.present? %>>
-      <%= context_href ? link_to(context_text, context_href, class: 'gem-c-title__context-link govuk-link', data: context_data) : context_text %>
+      <%= context %>
```

Comment:
To prevent a breaking change, could this be `context[:text]` with a warning saying that this will be deprecated in the future?

```suggestion
      <%= context.is_a?(Hash) ? context[:text] : context %>
```

Reply:
I considered that but it felt like leaving junk around to clear up later. I think I've captured all the places it's used and I'd rather be done with it in one go. I can be convinced if we have prior examples of handling change that way?

Reply:
That is a very good point. The prior example that comes to mind [is around Govspeak sanitisation][1] - but that warning message hung around for _years_, which I think makes a strong case that your method is better.

[1]: https://github.com/alphagov/govuk_publishing_components/pull/1632